File: app/src/main/java/piotr/michalkiewicz/mealplannerclient/user/model/UserSettings.java
package piotr.michalkiewicz.mealplannerclient.user.model;

import java.io.Serializable;
import java.util.List;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class UserSettings implements Serializable {

    private RecipeTypesSetting recipeTypesSetting;
    private NutritionProfileSettings nutritionProfileSettings;
    private Allergies allergies;
    private Diets diets;
    private List<String> unlikeIngredients;
    private List<String> eatenRecipes;
    private String Language;
    private int portionPreferences;
    private int cookingTimePreference;
    private int mealsPerMealPlanPreference;
}

File: app/src/main/java/piotr/michalkiewicz/mealplannerclient/user/model/UserSettings.kt
package piotr.michalkiewicz.mealplannerclient.user.model

import com.google.gson.annotations.SerializedName
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ApplicationComponent
import java.io.Serializable

@Module
@InstallIn(ApplicationComponent::class)
data class UserSettings(
    @SerializedName("nutritionProfileSettings") var nutritionProfileSettings: NutritionProfileSettings,
    @SerializedName("eatenRecipes") var eatenRecipes: List<String>,
    @SerializedName("language") var language: String,
    @SerializedName("location") var location: String?,
    @SerializedName("customizationDone") var customizationDone: Boolean,
    @SerializedName("sex") var sex: String?,
    @SerializedName("userPreferences") var userPreferences: UserPreferences?
) : Serializable